### DISCUSSION TITLE
Trajectory execution fails in MoveIt

### DISCUSSION THREAD

**Discussion by doshininad** [UniversalRobots/Universal_Robots_ROS2_Driver#472] [Q&A] [answered]
I have everything setup as described in the Readme.

I am connected to the real UR10 and when I launch the driver "ros2 launch ur_robot_driver ur_control_launch ur_type:=ur10e robot_ip:=192.168.1.102 launch_rviz:=true" it seems to be working fine. In Rviz I can visualize the current position of the real UR10. Also, on moving the UR10 manually I can see the position changing in Rviz in real time.

Then when I launch MoveIt "ros2 launch ur_moveit_config ur_moveit.launch.py ur_type:=ur10e launch_rviz:=true", I can see the real robot position in second Rviz screen along with start and goal positions of the robot with green and orange colors. The Plan command under Planning works fine and comes back with a planned trajectory but when I click Execute it fails. Also, the terminal shows and error from move_group-1 "Action client not connected to action server: joint_trajectory_controller/follow_joint_trajectory" "Failed to send trajectory part 1 of 1 to controller joint_trajectory_controller"

I know there is some silly mistake that I am doing, but I am not able to track it down. Can someone please help me. Thank You in advance.

**Comment by destogl** [comment]
Is joint trajectory controller running? Try to list your controllers using `ros2 control list_controllers`

**Reply by doshininad** [reply]
Thank you for your response.
Yes, when I set the initial_joint_controller = joint_trajectory_controller as an argument while started the launch file -- it worked. But, when I tried to set initial_joint_controller = scaled_joint_trajectory_controller, it doesn't seem to work. Any clues?

**Reply by fmauch** [reply]
To my knowledge MoveIt! will not determine which controller is actually running during runtime. We have both controllers configured inside https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/blob/main/ur_moveit_config/config/controllers.yaml, but it should be defaulting to `scaled_joint_trajectory_controller`. I can confirm, however, that it tries to execute the trajectory on `joint_trajectory_controller`, so I assume either an upstream bug or an upstream change that we need to adapt to in our configuration.

**Comment by fmauch** [comment, marked as answer]
This should be resolved by #480.

**Reply by doshininad** [reply]
Thank you for the solution.
So, now joint_trajectory_controller will be the default controller instead of scaled_joint_trajectory_controller?
Also, why is the real robot not able to execute the trajectory on scaled_joint_trajectory_controller?

**Reply by fmauch** [reply]
no, `scaled_joint_trajectory_controller` is the default controller unless specifying the `use_fake_hardware` flag.

**Reply by doshininad** [reply]
[2022-09-22 10-46-57.zip](https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/files/9626994/2022-09-22.10-46-57.zip)
This is what I am talking about. When I set the controller to scaled_joint_trajectory_controller, the trajectory execution on real robot fails but when I set it to joit_trajectory_controller, it works. Any clues why?

**Reply by fmauch** [reply]
But you did test that with the branch from #480?
